File: Components/WinForms/VB/DiagramMainDemo/Modules/LayoutModuleBase.vb
Imports System
Imports DevExpress.Diagram.Core
Imports DevExpress.XtraEditors
Imports System.Collections.Generic
Imports System.Windows.Forms
Imports DevExpress.Diagram.Core.Layout

Namespace DevExpress.XtraDiagram.Demos
    Partial Public Class LayoutModuleBase
        Inherits DiagramTutorialControl

        Protected Property IsLoaded() As Boolean

        Public Sub New()
            InitializeComponent()
            AddHandler diagramControl.RelayoutItemsCompleted, Sub(_o, _e) Diagram.FitToPage()
        End Sub
        Protected Overrides Sub OnLoad(ByVal e As EventArgs)
            MyBase.OnLoad(e)
            IsLoaded = True
            RelayoutDiagram()
        End Sub
        Protected Sub RelayoutDiagram()
            If Diagram Is Nothing OrElse (Not IsLoaded) Then
                Return
            End If

            RelayoutDiagramCore()
        End Sub
        Protected Overridable Sub RelayoutDiagramCore()
        End Sub
    End Class
End Namespace

File: Components/WinForms/CS/DiagramMainDemo/Modules/LayoutModuleBase.cs
using System;
using DevExpress.Diagram.Core;
using DevExpress.XtraEditors;
using System.Collections.Generic;
using System.Windows.Forms;
using DevExpress.Diagram.Core.Layout;

namespace DevExpress.XtraDiagram.Demos {
    public partial class LayoutModuleBase : DiagramTutorialControl {
        protected bool IsLoaded { get; set; }

        public LayoutModuleBase() {
            InitializeComponent();
            diagramControl.RelayoutItemsCompleted += (_o, _e) => Diagram.FitToPage();
        }
        protected override void OnLoad(EventArgs e) {
            base.OnLoad(e);
            IsLoaded = true;
            RelayoutDiagram();
        }
        protected void RelayoutDiagram() {
            if(Diagram == null || !IsLoaded) return;

            RelayoutDiagramCore();
        }
        protected virtual void RelayoutDiagramCore() { }
    }
}
